### PR TITLE
Adds git as a research topic

### PR DESCRIPTION
**Pull request by Karyum**
# Git
1. What is origin ? _eg. `git push origin master`_
2. How can you find the current origin ? and How can you change the origin's url ?
3. How to merge between 2 branches using `git merge` ?
4. different uses of `git-reset`.
5. Talk about or give a demo of the following commands:
    - `git stash` & `git stsah pop`
    - `git diff`
    - `git log`
    - `git fetch`

### CODE REVIEW

File: coursebook/week-2/research-afternoon.md [hunk 1]
```diff
@@ -22,15 +22,16 @@
...
-2. What are some basic accessibility checks that can be included in all of our projects? 
-3. Research how to:
-  + write an accessible navbar
-  + write an accessible modal  
-  (Hint: think about semantic HTML, managing focus, adding key handlers, adding aria attributes)
-4. Refactor one of your team's websites (or one from the rest of your cohort if necessary) to reflect what you've learned
+# Git
+1. What is origin ? _eg. `git push origin master`_
+2. How can you find the current origin ? and How can you change the origin's url ?
+3. How to merge between 2 branches using `git merge` ?
```
Comment (shiryz) [inline coursebook/week-2/research-afternoon.md]: I would add
```suggestion
3. How to merge between 2 branches using `git merge` ? 
  What would happen if you do `git merge master current-branch` 
```

File: coursebook/week-2/research-afternoon.md [hunk 2]
```diff
@@ -22,15 +22,16 @@
...
+4. different uses of `git-reset`.
+5. Talk about or give a demo of the following commands:
```
Comment (shiryz) [inline coursebook/week-2/research-afternoon.md]:
```suggestion
5. Talk about and give a demo of the following commands, when would you use each one?:
```

**Review by shiryz — CHANGES_REQUESTED** [review]

**Comment by shiryz** [comment]
@Karyum I've just remembered that we forgot `fork`

**Comment by shiryz** [comment]
I've also added the last research topic, sorry for hijacking this brach :sweat_smile:

File: coursebook/week-2/research-afternoon.md [hunk 3]
```diff
@@ -22,21 +22,25 @@
...
-## CSS
-1. What is responsive design? How does it compare to adaptive design?
-2. Compare the pros and cons of a desktop-first versus mobile-first approach to design. What do you think is generally preferable?
-3. What is BEM CSS and why does it matter?
-4. Compare the % and vw/vh units. What are the pros and cons of each? Where might you use each?
-5. Compare ems and rems as font size units. Why are they used and what would your advice be to someone choosing between the two?
-6. Refactor one of your team's websites to reflect what you've learned
+## Advanced JavaScript
+1. Value vs. reference types in JavaScript
+2. `==` vs. `===`
```
Comment (Karyum) [inline coursebook/week-2/research-afternoon.md:39]: isn't this too easy ? 

we have a giant list of javascript concepts to choose from

File: coursebook/week-2/research-afternoon.md [hunk 4]
```diff
@@ -22,21 +22,25 @@
...
+3. Scope
+    - What are the different types of scopes 
+    - How does the scope affect variables and variable decleration
+4. The `this` keyword
```
Comment (Karyum) [inline coursebook/week-2/research-afternoon.md]: i don't think this is a good idea, the `this` keyword is useless unless youre using classes, cause it's considered bad practice to use it in `function` or an object

File: coursebook/week-2/research-afternoon.md (hunk 3 above)
Comment (shiryz) [reply coursebook/week-2/research-afternoon.md:39]: They should understand the difference and it relates to the previous point

File: coursebook/week-2/research-afternoon.md (hunk 4 above)
Comment (shiryz) [reply coursebook/week-2/research-afternoon.md]: This relates to the scope as well but I can remove because it can be really confusing

**Review by shiryz — APPROVED** [review]

**Pull request merged**